modules/DesktopProject/src/main/java/org/gephi/desktop/project/ProjectControllerUIImpl.java

```diff
@@ -222,22 +226,28 @@ private void updateTitleBar(Project project) {
         });
     }
 
-    private void saveProject(Project project, File file) {
-        longTaskExecutor.execute(null, () -> {
-            controller.saveProject(project, file);
+    private Future<Void> saveProject(Project project, File file) {
+
+
+        return longTaskExecutor.execute(null,   new Callable<Void>() {
+            @Override
+            public Void call() throws Exception {
+                controller.saveProject(project, file);
+                return null;
+            }
         });
     }
 
-    public void saveProject() {
+    public Future<Void> saveProject() {

```

Comment:
Actually @mbastian Should this function block ? So that in any case we block on save ?